go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/mongodb/terraform-provider-mongodbatlas
 
-go 1.20
+go 1.21
```

Comment:
I'm not sure we should upgrade to Go 1.21 until we decide so, i don't know that is the process or how time we normally wait until we consider a version stable